integrasjonspunkt/src/main/java/no/difi/meldingsutveksling/nextmove/NextMoveServiceBus.java

```diff
@@ -49,34 +52,40 @@
     private static final String NEXTMOVE_QUEUE_PREFIX = "nextbestqueue";
 
     private IntegrasjonspunktProperties props;
-    private ServiceRegistryLookup sr;
     private StandardBusinessDocumentFactory sbdf;
     private MessageSender messageSender;
     private NextMoveQueue nextMoveQueue;
     private JAXBContext jaxbContext;
-    private String queuePath;
+    private ServiceBusRestClient serviceBusClient;
+    private IMessageReceiver messageReceiver;
 
     @Autowired
     public NextMoveServiceBus(IntegrasjonspunktProperties props,
                               StandardBusinessDocumentFactory sbdf,
-                              ServiceRegistryLookup sr,
                               MessageSender messageSender,
-                              NextMoveQueue nextMoveQueue) throws JAXBException {
+                              NextMoveQueue nextMoveQueue,
+                              ServiceBusRestClient serviceBusClient) throws JAXBException {
         this.props = props;
         this.sbdf = sbdf;
-        this.sr = sr;
         this.messageSender = messageSender;
         this.nextMoveQueue = nextMoveQueue;
+        this.serviceBusClient = serviceBusClient;
         this.jaxbContext = JAXBContextFactory.createContext(new Class[]{EduDocument.class, Payload.class, ConversationResource.class}, null);
-        queuePath = format("%s%s%s", NEXTMOVE_QUEUE_PREFIX,
-                props.getOrg().getNumber(),
-                props.getNextbest().getServiceBus().getMode());
     }
 
-    public void putMessage(ConversationResource resource) throws NextMoveException {
+    @PostConstruct
+    public void init() throws ServiceBusException, InterruptedException {
```

Comment:
![MAJOR](https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR') Refactor this method to throw at most one checked exception instead of: com.microsoft.azure.servicebus.primitives.ServiceBusException, java.lang.InterruptedException [![rule](https://sonarsource.github.io/sonar-github/rule.png)](http://sonar-app01.difi.local:9000/coding_rules#rule_key=squid%3AS1160)